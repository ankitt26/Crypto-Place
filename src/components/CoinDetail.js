import { Link, useParams } from 'react-router-dom';
import { IoChevronBackSharp } from 'react-icons/io5';
import { useSelector } from 'react-redux';
import styles from '../CSS/CoinDetail.module.css';

const CoinDetail = () => {
  const { coinId } = useParams();
  const coindata = useSelector((state) => state.Coins.CoinList);
  const coinInfo = coindata.filter((element) => element.id === coinId);

  const renderCoinInfo = coinInfo.map((element) => (
    <main className={styles.card} key={element.id}>
      <Link to="/" className={styles.back_btn}>
        <IoChevronBackSharp />
      </Link>
      <div className={styles.header}>
        <h2 className={styles.coin}>{element.name}</h2>
      </div>
      <img className={styles.img} src={element.image} alt={element.name} />
      <table className={styles.detail_table}>
        <thead className={styles.thead}>
          <tr>
            <th className={styles.th1}>Data</th>
            <th className={styles.th2}>Value</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className={styles.td1}>Current Price</td>
            <td className={styles.td2}>
              $
              {element.current_price}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Market Rank</td>
            <td className={styles.td2}>{element.market_cap_rank}</td>
          </tr>
          <tr>
            <td className={styles.td1}>Total Volume</td>
            <td className={styles.td2}>
              $
              {element.total_volume}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>High 24h</td>
            <td className={styles.td2}>
              $
              {element.high_24h}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Low 24h</td>
            <td className={styles.td2}>
              $
              {element.low_24h}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Price Change 24h</td>
            <td className={styles.td2}>
              $
              {element.price_change_24h}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Price Change Percentage 24h</td>
            <td className={styles.td2}>
              {' '}
              {element.price_change_percentage_24h}
              {' '}
              %
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Market Cap Change 24h</td>
            <td className={styles.td2}>
              $
              {element.market_cap_change_24h}
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Market Cap Change Percentage 24h</td>
            <td className={styles.td2}>
              {' '}
              {element.market_cap_change_percentage_24h}
              {' '}
              %
            </td>
          </tr>
          <tr>
            <td className={styles.td1}>Last Updated</td>
            <td className={styles.td2}>
              {' '}
              {element.last_updated}
            </td>
          </tr>
        </tbody>
      </table>
    </main>
  ));

  return <>{renderCoinInfo}</>;
};

export default CoinDetail;
