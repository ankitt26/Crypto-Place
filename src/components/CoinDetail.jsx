import { Link, useParams,} from 'react-router-dom';
import { IoChevronBackSharp } from 'react-icons/io5';
import styles from "../CSS/CoinDetail.module.css";
const CoinDetail = () => {
  const {coin_id} = useParams();
   console.log(coin_id);
    return (
    <main className={styles.card} >
      <div className={styles.header}>
        <Link to="/" className={styles.back_btn}> {<IoChevronBackSharp/>} </Link>
        <h2 className={styles.coin}>Bitcoin</h2>
      </div>
      <img  className={styles.img} src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"></img>
      <div className={styles.detail_box}>
        <div className={styles.box1}>
          <h3>current_price </h3>
          <h3> market_cap</h3>
          <h3>total_volume</h3>
          <h3>high_24h</h3>
          <h3>low_24h</h3>
          <h3>price_change_24h</h3>
          <h3>price_change_24h</h3>
          <h3>market_cap_change_24h</h3>
          <h3>market_cap_change_24h</h3>
          <h3>price_change_1h</h3>
          <h3> price_change_24h</h3>
          <h3>price_change_7d</h3>
          <h3>last_updated</h3>
        </div>
        <div className={styles.box2}>
          <h3>548 </h3>
          <h3> 51165</h3>
          <h3>156</h3>
          <h3>561</h3>
          <h3>6516</h3>
          <h3>6515</h3>
          <h3>5115</h3>
          <h3>6516</h3>
          <h3>651615</h3>
          <h3>651</h3>
          <h3> 51</h3>
          <h3>51</h3>
          <h3>6511161611</h3>
        </div>
      </div>
    </main>
  );
};

export default CoinDetail;