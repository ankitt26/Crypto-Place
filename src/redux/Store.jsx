import { configureStore } from '@reduxjs/toolkit';
import CoinSlice from './Coins/CoinSlice';
import SearchSlice from './Coins/SearchSlice';
const Store = configureStore({
    reducer: {
        Coins : CoinSlice,
        Search : SearchSlice
    },
})

export default Store ;