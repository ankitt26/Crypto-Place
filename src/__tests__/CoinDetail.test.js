import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CoinDetail from '../components/CoinDetail';
import Store from '../redux/Store';

it('coindetail test', () => {
  const component = renderer.create(
    <Provider store={Store}>
      <CoinDetail />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
