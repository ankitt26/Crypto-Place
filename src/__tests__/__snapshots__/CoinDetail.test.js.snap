// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coindetail test 1`] = `null`;
